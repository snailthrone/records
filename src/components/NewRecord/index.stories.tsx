import React from 'react';
import {Â Meta, Story } from '@storybook/react/types-6-0';

import NewRecord, { Props } from '.';

export default { title: 'Components/NewRecord', component: NewRecord } as Meta;

const Template: Story<Props> = args => <NewRecord {...args} />

export const Default = Template.bind({});
