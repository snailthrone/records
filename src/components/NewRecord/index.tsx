import React, { FC, useState } from 'react';

export type Props = unknown;

const NewRecord: FC<Props> = () => {
  const [album, setAlbum] = useState('');
  const [artist, setArtist] = useState('');
  const [year, setYear] = useState('');
  const [catalogId, setCatalogId] = useState('');

  const submit = async event => {
    event.preventDefault();

    const newAlbum = { album, artist, year, catalogId };

    console.log(newAlbum)

    setAlbum('');
    setArtist('');
    setYear('');
    setCatalogId('');
  }

  return (
    <div>
      <h3>Add new record</h3>
      <form onSubmit={submit}>
        <div>
          <label>
            Album:
            {' '}
            <input onChange={({ target: { value }}) => setAlbum(value)} type="text" value={album} />
          </label>
        </div>
        <div>
          <label>
            Artist:
            {' '}
            <input onChange={({ target: { value }}) => setArtist(value)} type="text" value={artist} />
          </label>
        </div>
        <div>
          <label>
            Year:
            {' '}
            <input onChange={({ target: { value }}) => setYear(value)} type="text" value={year} />
          </label>
        </div>
        <div>
          <label>
            Catalog id:
            {' '}
            <input onChange={({Â target: { value }}) => setCatalogId(value)} type="text" value={catalogId} />
          </label>
        </div>
        <button type="submit">Add</button>
      </form>
    </div>
  )
};

export default NewRecord;
