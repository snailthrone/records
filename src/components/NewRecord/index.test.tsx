import React from 'react';
import renderer from 'react-test-renderer';
import NewRecord from './index';

describe('New Record', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NewRecord />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
