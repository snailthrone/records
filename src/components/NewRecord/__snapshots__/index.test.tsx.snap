// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Record renders correctly 1`] = `
<div>
  <h3>
    Add new record
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label>
        Album:
         
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        Artist:
         
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        Year:
         
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        Catalog id:
         
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
